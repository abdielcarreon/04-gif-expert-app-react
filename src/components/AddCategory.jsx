import { useState } from "react"
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {

    const [ inputValue, setInputValue ] = useState('');
    
    const onInputChange = (  event  ) => {
        setInputValue(event.target.value)
    }

    const onSubmit = (event) => {
        console.log('Hello from Submit')
        event.preventDefault();
        const trimCategory = inputValue.trim()
        if(trimCategory.length <= 1) return;
        //setCategories(categories =>[ inputValue,...categories ]);
        onNewCategory(trimCategory);
        setInputValue('');   
    }

  return (
    <form onSubmit={ event => onSubmit(event) }
    aria-label="form">
        <input
            type="text"
            placeholder={"Search gifs"}
            value={ inputValue }
            onChange={ onInputChange }
        />
    </form>
    
  )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}