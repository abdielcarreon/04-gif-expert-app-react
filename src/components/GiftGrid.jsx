
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem } from './GifItem';
import  PropTypes  from 'prop-types';

export const GiftGrid = ({ category }) => {
    
    const { images, isLoading } = useFetchGifs( category );

    console.log({ images, isLoading });
    

    return (
        <>
            <h3>{ category } </h3>
            {
                isLoading && ( <h2>Loading...</h2> )
            }
            

            <div className='card-grid'>
                {
                    images.map( ( image ) => (
                        <GifItem key={ image.id }  
                        { ...image }
                        />
                    ))
                }
            </div>
        </>
    )
}

GiftGrid.propTypes = {
    category: PropTypes.string.isRequired
}