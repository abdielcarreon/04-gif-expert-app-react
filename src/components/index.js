

export * from './AddCategory';
export * from './AddCategory';
export * from './GiftGrid';