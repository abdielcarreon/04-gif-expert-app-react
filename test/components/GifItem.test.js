import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem"



describe('Testing on <GiftItem />', () => {

  const title = 'Saitama';
  const url = 'https://one-punch.com/saitama.jpg'
  test('should to match snapshot', () => {
    
      const { container } = render(< GifItem title={ title } url= { url } />)
    
      expect( container ).toMatchSnapshot();

  });

  test('should show the image with url and alt indicated', () => {
    
    const { container } = render(< GifItem title={ title } url= { url } />)

    //screen.debug();

    // expect(screen.getByRole('img').src).toBe( url );

    // expect(screen.getByRole('alt').src).toBe( title );

    const { src, alt } = screen.getByRole('img')

    expect( src ).toBe( url )
    expect( alt ).toBe( title )
  });

  test('should show the title in the component', () => {
    
    const { container } = render(< GifItem title={ title } url= { url } />);

    expect( screen.getByText( title ) ).toBeTruthy();

  })
  
});
