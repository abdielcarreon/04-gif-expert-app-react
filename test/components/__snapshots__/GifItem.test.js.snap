// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <GiftItem /> should to match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;

exports[`Testing on <GiftItem /> should to match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
